### ISSUE TITLE
Read Indicator for the Last Message Sent

### ISSUE THREAD

**Issue by srkunze** [qTox/qTox#1083]
Is there something like read confirmations built-in in (q)Tox?

After finally getting Antox working on her tablet, my girlfriend (again ;) ) said it would be nice to have a read indicator for situations where she only wants to inform others and make sure the other one has received and read the piece of information. For instance for arranging appointments.

**Comment by dubslow** [comment]
Read failure indicators are a thing, and it may be better visually with the new chatform.

**Comment by ProMcTagonist** [comment]
Are there read receipts in core?

**Comment by dubslow** [comment]
Yes, and I believe qTox implements them. How they're displayed in the GUI, I'm not so sure. It's likely better in the new chat form.

**Comment by zetok** [comment]
Only part about receiving message is confirmed. Software has no real way of knowing whether someone has actually read something…

@srkunze qTox has support for it. To see how it looks when qTox doesn't get confirmation of message being received, just send message to offline friend.

About change to how they're displayed; instead of that red thing you'll get a circle spinning as long as qTox won't get a confirmation of message being delivered.

**Comment by zetok** [comment]
@srkunze For Antox related stuff you should fill issues for them in a proper place: https://github.com/subliun/Antox/issues

This issue can be closed, I think, since qTox already confirms delivery, and there's no way to know whether someone read message. Theoretically something about this could be done, but it would have to be opt-in for every user (due to privacy reasons), so in practice almost no one would have it turned on anyway…

**Comment by srkunze** [comment]
> Only part about receiving message is confirmed. Software has no real way of knowing whether someone has actually read something…

As I said, 'normal' people might consider it useful. Facebook doesn't know it either, but they try it.

> qTox already confirms delivery

I know but delivery is **technical**. This issue is about what's important to humans. :)

I understand that this could be "implemented" by good habits as well i.e. sending a receipt like "Got it." or "Alright" as soon as one has read the message. Not sure if that is sufficient. But I guess the **convenience** is that those messages are sent as soon as I have the chat window open for more than, say, 3 seconds maybe depending on the length of the message.

**Comment by agilob** [comment]
Done?

**Comment by srkunze** [comment]
@agilob Whom are you asking?

**Comment by aaannndddyyy** [comment]
whatsapp has it. i don't know if we need it though, and am pretty sure that core does not have that feature.

**Comment by zetok** [comment]
I think that it might be nice to have it as an opt-in feature. (note though that implementing that most likely would have very low priority)

This requires support in toxcore, so an issue should be opened there.

As for when message would be marked as `read` - when someone would have it open in their client.

**Comment by aaannndddyyy** [comment]
+1

It's a nice fearurw to have. Not conplicate to do. And many users expect it.
Should thereforw be enabled by default, but of course there must be an optipn not tp send it - just like we have withe typing notifications.null

**Comment by ProMcTagonist** [comment]
+1 @zetok
